JS/Lesson35/savas/my-app/src/components/RecipeDetail/index.jsx

```diff
@@ -1,12 +1,8 @@
-import * as React from 'react';
-import { useParams } from 'react-router-dom';
-
+import * as React from "react";
+import { useParams } from "react-router-dom";
 
 export const RecipeDetail = () => {
+  let { id } = useParams();
```

Comment:
You mentioned that you had a problem fetching recipe details in this component. It can be done like this:
```
	const [recipe, setRecipe] = useState();

	useEffect(() => {
		if (id) {
			fetch(`https://www.themealdb.com/api/json/v1/1/lookup.php?i=${id}`)
				.then((response) => response.json())
				.then((data) => setRecipe(data.meals[0]));
		}
	}, [id]);
```

However, this approach is not perfect as it makes additional API request. Instead, when we app ContextAPI for recipes or Redux store, we should fetch recipe data from the store.

Reply:
Running the codes, I get an error like 'No routes matched location "/recipes/52948"'

Reply:
There're are few issues. 
1. Route for Recipe details page is not added correctly, change AppRouter.jsx:
`      <Route path="/recipes/:id" element={<RecipeDetail />} />
`
2. Don't pass id as prop to RecipeDetail component. We cannot pass it as a prop any more as we create a link to the  component dynamically. 
This is how we did it in the lesson: https://github.com/frontend-from-0/group240123-files/blob/master/Lesson35/my-app/src/components/RecipeDetail/index.jsx but below I changed the code a bit more to acommodate for situations where recipe is not yet available 
```
import * as React from 'react';
import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';

export const RecipeDetail = () => {
	const { id } = useParams();
	const [recipe, setRecipe] = useState();

	useEffect(() => {
		if (id) {
			fetch(`https://www.themealdb.com/api/json/v1/1/lookup.php?i=${id}`)
				.then((response) => response.json())
				.then((data) => setRecipe(data.meals[0]));
		}
	}, [id]);

	return recipe ? (
		<div>Recipe Detail: {recipe.strMeal}</div>
	) : (
		<div>Loading...</div>
	);
};
```